pkg/cmd/job/view/view.go

```diff
@@ -0,0 +1,246 @@
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	JobID      string
+	Log        bool
+	ExitStatus bool
+
+	Prompt bool
+
+	Now func() time.Time
+}
+
+func NewCmdView(f *cmdutil.Factory, runF func(*ViewOptions) error) *cobra.Command {
+	opts := &ViewOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		Now:        time.Now,
+	}
+	cmd := &cobra.Command{
+		Use:    "view [<job-id>]",
+		Short:  "View the summary or full logs of a workflow run's job",
+		Args:   cobra.MaximumNArgs(1),
+		Hidden: true,
+		Example: heredoc.Doc(`
+			# Interactively select a run then job
+			$ gh job view
+
+			# Just view the logs for a job
+			$ gh job view 0451 --log
+
+			# Exit non-zero if a job failed
+			$ gh job view 0451 -e && echo "job pending or passed"
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+
+			terminal := opts.IO.IsStdoutTTY() && opts.IO.IsStdinTTY()
+
+			if len(args) > 0 {
+				opts.JobID = args[0]
+			} else if !terminal {
+				return &cmdutil.FlagError{Err: errors.New("job ID required when not running interactively")}
+			} else {
+				opts.Prompt = true
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return runView(opts)
+		},
+	}
+	cmd.Flags().BoolVarP(&opts.Log, "log", "l", false, "Print full logs for job")
+	// TODO should we try and expose pending via another exit code?
+	cmd.Flags().BoolVarP(&opts.ExitStatus, "exit-status", "e", false, "Exit with non-zero status if job failed")
+
+	return cmd
+}
+
+func runView(opts *ViewOptions) error {
+	c, err := opts.HttpClient()
+	if err != nil {
+		return fmt.Errorf("failed to create http client: %w", err)
+	}
+	client := api.NewClientFromHTTP(c)
+
+	repo, err := opts.BaseRepo()
+	if err != nil {
+		return fmt.Errorf("failed to determine base repo: %w", err)
+	}
+
+	out := opts.IO.Out
+	cs := opts.IO.ColorScheme()
+
+	jobID := opts.JobID
+	if opts.Prompt {
+		runID, err := shared.PromptForRun(cs, client, repo)
+		if err != nil {
+			return err
+		}
+		// TODO I'd love to overwrite the result of the prompt since it adds visual noise but I'm not sure
+		// the cleanest way to do that.
+		fmt.Fprintln(out)
+
+		opts.IO.StartProgressIndicator()
+
+		run, err := shared.GetRun(client, repo, runID)
+		if err != nil {
+			return fmt.Errorf("failed to get run: %w", err)
+		}
+
+		opts.IO.StopProgressIndicator()
+
+		jobID, err = promptForJob(*opts, client, repo, *run)
+		if err != nil {
+			return err
+		}
+
+		fmt.Fprintln(out)
+	}
+
+	opts.IO.StartProgressIndicator()
+
+	job, err := getJob(client, repo, jobID)
+	if err != nil {
+		return fmt.Errorf("failed to get job: %w", err)
+	}
+
+	opts.IO.StopProgressIndicator()
+
+	if opts.Log {
+		opts.IO.StartProgressIndicator()
+		r, err := client.JobLog(repo, jobID)
+		if err != nil {
+			return err
+		}
+		opts.IO.StopProgressIndicator()
+
+		if _, err := io.Copy(out, r); err != nil {
```

Comment:
Could we enable the pager for output? This filled several screenfuls in my terminal and I really wish it went to my configured pager instead.